lib/Scheduler/machine_model.cpp

```diff
@@ -17,6 +17,31 @@ using std::vector;
 MachineModel::MachineModel(const std::string &modelFile) {
   SpecsBuffer buf;
   buf.Load(modelFile.c_str());
+
+  char buffer[MAX_NAMESIZE];
+
+  buf.ReadSpec("MODEL_NAME:", buffer);
+  mdlName_ = buffer;
+
+  issueRate_ = buf.ReadIntSpec("ISSUE_RATE:");
+
+  int numIssueTypes = buf.ReadIntSpec("ISSUE_TYPE_COUNT:");
+
+  issueTypes_.resize(numIssueTypes > 0 ? numIssueTypes : 1);
+  if (numIssueTypes > 0) {
+    for (size_t j = 0; j < issueTypes_.size(); j++) {
+      int pieceCnt;
+      char *strngs[INBUF_MAX_PIECES_PERLINE];
+      int lngths[INBUF_MAX_PIECES_PERLINE];
+      buf.GetNxtVldLine(pieceCnt, strngs, lngths);
+
+      if (pieceCnt != 2)
+        llvm::report_fatal_error("Invalid issue type spec", false);
+
+      issueTypes_[j].name = strngs[0];
+      issueTypes_[j].slotsCount = atoi(strngs[1]);
+    }
+  }
```

Comment:
Based on the change that introduced this bug (#162), I think this should include all the other code of the other constructor as well.

Rather than duplicate the code below, I would suggest pulling the other constructor's body into a helper method and having this constructor call into it:

```c++
// Member of MachineModel:
private:
  void parseBuffer(SpecsBuffer &buf);

// In machine_model.cpp:
void MachineModel::parseBuffer(SpecsBuffer &buf) {
  // The other constructor body.
}

// The two constructors would forward to that:
MachineModel::MachineModel(const std::string& modelFile) {
  SpecsBuffer buf;
  buf.Load(modelFile.c_str());
  parseBuffer(buf);
}

MachineModel::MachineModel(SpecsBuffer &buf) {
  parseBuffer(buf);
}
```

Reply:
I'm not sure why I didn't see these comments when you posted -- sorry about that. Anyway, I agree with this design. I'll update the commit